apps/src/javalab/JavalabEditor.jsx

```diff
@@ -807,61 +739,21 @@ class JavalabEditor extends React.Component {
             </Tab.Content>
           </div>
         </Tab.Container>
-        <JavalabDialog
-          isOpen={editorOpenDialogName === JavalabEditorDialog.DELETE_FILE}
-          handleConfirm={this.onDeleteFile}
-          handleClose={() => closeEditorDialog()}
-          message={javalabMsg.deleteFileConfirmation({
-            filename: fileMetadata[fileToDelete]
-          })}
-          displayTheme={displayTheme}
-          confirmButtonText={javalabMsg.delete()}
-          closeButtonText={javalabMsg.cancel()}
-        />
-        <NameFileDialog
-          isOpen={editorOpenDialogName === JavalabEditorDialog.RENAME_FILE}
-          handleClose={() => {
-            closeEditorDialog();
-            this.setState({renameFileError: null});
-          }}
-          filename={fileMetadata[editTabKey]}
-          handleSave={this.onRenameFile}
-          displayTheme={displayTheme}
-          inputLabel="Rename the file"
-          saveButtonText="Rename"
-          errorMessage={renameFileError}
-        />
-        <NameFileDialog
-          isOpen={editorOpenDialogName === JavalabEditorDialog.CREATE_FILE}
-          handleClose={() => {
-            closeEditorDialog();
-            this.setState({newFileError: null});
-          }}
-          handleSave={this.onCreateFile}
-          displayTheme={displayTheme}
-          inputLabel="Create new file"
-          saveButtonText="Create"
-          errorMessage={newFileError}
-          filename={DEFAULT_FILE_NAME}
-        />
-        <CommitDialog
-          isOpen={editorOpenDialogName === JavalabEditorDialog.COMMIT_FILES}
-          files={Object.keys(sources)}
-          handleClose={() => {
-            closeEditorDialog();
-          }}
-          handleCommit={onCommitCode}
+        <JavalabEditorDialogManager
+          onDeleteFile={this.onDeleteFile}
+          filenameToDelete={fileMetadata[fileToDelete]}
+          onRenameFile={this.onRenameFile}
+          filenameToRename={fileMetadata[editTabKey]}
```

Comment:
Not sure if this would make it more confusing, but I think this could come from Redux

Reply:
Ah good call, will update to use Redux here.

Reply:
Actually looking at this again, looks like `fileToDelete` is part of JavalabEditor's state so we'd still need to pass it in somehow. I could move the `filenameToRename` to redux since `editTabKey` comes from redux, but I think I may keep it as is for now for consistency, if that sounds ok? I think once we merge all these we'll be able to still move more stuff into redux/separate components so this would end up simplifying anyway.

Reply:
Yep I was looking specifically at `filenameToRename`, and yeah agree consistency sounds good for now 👍